data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessorConfig.java

```diff
@@ -138,13 +157,16 @@ public class KeyValueProcessorConfig {
             "The default configuration is <code>false</code> which retains those grouping characters.")
     private boolean removeBrackets;
 
-    @JsonProperty(value = "value_grouping", defaultValue = "false")
+    @JsonProperty(value = VALUE_GROUPING_KEY, defaultValue = "false")
     @JsonPropertyDescription("Specifies whether to group values using predefined grouping delimiters. " +
             "If this flag is enabled, then the content between the delimiters is considered to be one entity and " +
             "they are not parsed as key-value pairs. The following characters are used a group delimiters: " +
             "<code>{...}</code>, <code>[...]</code>, <code>&lt;...&gt;</code>, <code>(...)</code>, <code>\"...\"</code>, <code>'...'</code>, <code>http://... (space)</code>, and <code>https:// (space)</code>. " +
             "Default is <code>false</code>. For example, if <code>value_grouping</code> is <code>true</code>, then " +
             "<code>{\"key1=[a=b,c=d]&amp;key2=value2\"}</code> parses to <code>{\"key1\": \"[a=b,c=d]\", \"key2\": \"value2\"}</code>.")
+    @AlsoRequired(values = {
+            @AlsoRequired.Required(name = FIELD_DELIMITER_REGEX_KEY, allowedValues = {"null"})
+    })
     private boolean valueGrouping = false;
 
     @JsonProperty(value = "recursive", defaultValue = "false")
```

Comment:
recursive config has a restriction where remove_brackets must be false, skip_duplicate_values is always true, and whitespace is always strict